src/components/App.jsx

```diff
@@ -1,11 +1,158 @@
 import React from 'react'
-import ContactsApp from './ContactsApp'
-import { Layout } from 'cozy-ui/react/Layout'
+import ContactsList from './ContactsList'
+import withSelection from './HOCs/withSelection'
+import { PropTypes } from 'prop-types'
+import ContactCardModal from './Modals/ContactCardModal'
+import ContactFormModal from './Modals/ContactFormModal'
+import { Alerter } from 'cozy-ui/react'
+import { Main, Content, Layout } from 'cozy-ui/react/Layout'
+import connect from '../connections/allContacts'
+import { getFullContactName } from '../helpers/contacts'
+import ContactImportationModal from './ContactImportationModal'
+import Header from './Header'
+import Toolbar from './Toolbar'
+import ContactsSelectionBar from './layout/ContactsSelectionBar'
 
-const App = () => (
-  <Layout monocolumn>
-    <ContactsApp />
-  </Layout>
-)
+class ContactsApp extends React.Component {
+  state = {
+    displayedContact: null,
+    isImportationDisplayed: false,
+    isCreationFormDisplayed: false
+  }
 
-export default App
+  displayImportation = () => {
+    this.setState({
+      isImportationDisplayed: true
+    })
+  }
+
+  hideImportation = () => {
+    this.setState({
+      isImportationDisplayed: false
+    })
+  }
+
+  displayContactCard = contact => {
+    this.setState({
+      displayedContact: contact
+    })
+  }
+
+  onDeleteContact = contact => {
+    this.hideContactCard()
+    Alerter.info('delete-confirmation.deleted', {
+      name: getFullContactName(contact.name)
+    })
+  }
+
+  hideContactCard = () => {
+    this.setState({
+      displayedContact: null
+    })
+  }
+
+  displayContactForm = () => {
+    this.setState({
+      isCreationFormDisplayed: true
+    })
+  }
+
+  hideContactForm = () => {
+    this.setState({
+      isCreationFormDisplayed: false
+    })
+  }
+
+  onCreateContact = contact => {
+    this.hideContactForm()
+    this.displayContactCard(contact)
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.state.displayedContact) {
+      this.setState(state => ({
+        ...state,
+        displayedContact: prevProps.contacts.find(
+          contact => contact._id === state.displayedContact._id
+        )
+      }))
+    }
+  }
+
+  render() {
+    const {
+      displayedContact,
+      isImportationDisplayed,
+      isCreationFormDisplayed
+    } = this.state
+    const { t } = this.context
+
+    return (
+      <Layout monocolumn>
+        <Main>
+          <ContactsSelectionBar
+            selection={this.props.selection}
```

Comment:
Maybe we could extract these props at the start of the render function? It would be consistent with what we do with `context` and `state`, and on a more personal level I find it very practical to see at a glance all the props that will be used.

Reply:
It's a matter of personal taste: I prefer to expose the props drilling.